packages/api/internal/handlers/sandbox_get.go

```diff
@@ -57,35 +62,45 @@ func (a *APIStore) GetSandboxesSandboxID(c *gin.Context, id string) {
 	}
 
 	// If sandbox not found try to get the latest snapshot
-	snapshot, build, err := a.db.GetLastSnapshot(ctx, sandboxId, team.ID)
+	lastSnapshot, err := a.sqlcDB.GetLastSnapshot(ctx, queries.GetLastSnapshotParams{SandboxID: sandboxId, TeamID: team.ID})
 	if err != nil {
 		zap.L().Error("error getting last snapshot for sandbox", zap.String("sandbox_id", id), zap.Error(err))
 		c.JSON(http.StatusNotFound, fmt.Sprintf("sandbox \"%s\" doesn't exist or you don't have access to it", id))
 		return
 	}
 
-	memoryMB := int32(-1)
-	cpuCount := int32(-1)
+	memoryMB := int32(lastSnapshot.EnvBuild.RamMb)
+	cpuCount := int32(lastSnapshot.EnvBuild.Vcpu)
+
+	var sbxAccessToken *string = nil
+	if lastSnapshot.Snapshot.EnvSecure {
+		hashed := sandbox.NewEnvdAccessTokenGenerator()
```

Comment:
is it necessary to init the generator for each sandbox, can't there be a global generator?